File: DesignPattern/DesignPattern/Decorator/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DesignPattern.Decorator
{
    class Person
    {
        public Person() { }

        private string m_name;
        public Person(string name)
        {
            m_name = name;
        }

        public virtual void Show()
        {
            Console.WriteLine($"妆扮的{m_name}");
        }
    }
}

File: DesignPattern/DesignPatternGo/decorators/person.go
package decorators

type Person struct {
	name   string
	finery Finery
}

func (p Person) Print() {
	print(p.name + " 穿着 " + p.finery.Show())
}

func NewPerson(name string, f Finery) Person {
	return Person{
		name:   name,
		finery: f,
	}
}
